presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1449,6 +1460,9 @@ private RelationType analyzeView(Query query, QualifiedObjectName name, Optional
             Scope queryScope = analyzer.process(query, Scope.create());
             return queryScope.getRelationType().withAlias(name.getObjectName(), null);
         }
+        catch (SemanticException e) {
+            throw e;
```

Comment:
Well, without this line it just means that user will get a VIEW_ANALYSIS_ERROR but the error message will still say that view is recursive. Both seems reasonable to me so I'll let you decide.